### ISSUE TITLE
CSI snapshot integration

### ISSUE THREAD

**Issue by skriss** [vmware-tanzu/velero#2066]

**Comment by skriss** [comment]
we've got 3 issues under this epic, but we need to break this down further. To do as a team once we finalize design.

**Comment by dns2utf8** [comment]
What is the status of this? Is it possible to backup the contents of a `generic-nimble-csi-2` with version 1.2.0?

**Comment by skriss** [comment]
@dns2utf8 in v1.2, you can back up non-CSI volumes (e.g. EBS, Google PD, Azure Disk) using snapshots, or any other type of volume using restic (https://velero.io/docs/v1.2.0/restic/).

This issue covers integrating with the new CSI snapshot API, to be able to take snapshots of CSI-provisioned volumes.

**Comment by dns2utf8** [comment]
Do I understand correctly that this means CSI volumes are not snapshoted as of today?
I would like to backup the PVs of a MariaDB and a PostgreSQL.

**Comment by skriss** [comment]
@dns2utf8 that is correct. As of today, Velero can snapshot non-CSI volumes, or use restic to back up CSI volumes.

**Comment by dns2utf8** [comment]
Thank you for your replies. I had some initial success with restic.
However, now the contents of the PV don't get into the s3 bucket anymore and the pods are stuck at the "Terminating" state.
I searched the web for a solution, is the `MountPropagation` still needed with the `securityContext: privileged: true`?

**Comment by nrb** [comment]
Currently waiting on upstream providers to add v2.0.0 controller/v1beta1 CRD support to their drivers. I've [logged an issue on the Amazon driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/442) to start.

While we could proceed with the v1alpha1 code, I'd prefer to keep moving forward and help update drivers rather than move backward and end up updating a lot of code later.

**Comment by ashish-amarnath** [comment]
This is the issue with the GCP driver: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/457

**Comment by ashish-amarnath** [comment]
There are 3 more issues pending in this epic. https://github.com/vmware-tanzu/velero/labels/CSI
Which are not going to make it for 1.5.
Punting the remaining issues to 1.6.
